scipy/stats/_distn_infrastructure.py

```diff
@@ -3251,6 +3202,130 @@ def _iter_chunked(x0, x1, chunksize=4, inc=1):
         yield supp
 
 
+class rv_sample(rv_discrete):
+    """A 'sample' discrete distribution defined by the support and values.
+
+       The ctor ignores most of the arguments, only needs the `values` argument.
+    """
+    def __init__(self, a=0, b=inf, name=None, badvalue=None,
+                 moment_tol=1e-8, values=None, inc=1, longname=None,
+                 shapes=None, extradoc=None, seed=None):
+
+        super(rv_discrete, self).__init__(seed)
+
+        if values is None:
+            raise ValueError("rv_sample.__init__(..., values=None,...)")
+
+        # cf generic freeze
+        self._ctor_param = dict(
+            a=a, b=b, name=name, badvalue=badvalue,
+            moment_tol=moment_tol, values=values, inc=inc,
+            longname=longname, shapes=shapes, extradoc=extradoc, seed=seed)
+
+        if badvalue is None:
+            badvalue = nan
+        if name is None:
+            name = 'Distribution'
+        self.badvalue = badvalue
+        self.name = name
+        self.moment_tol = moment_tol
+        self.inc = inc
+        self.shapes = shapes
+        self.extradoc = extradoc
+
+        self.xk, self.pk = values
+        self.return_integers = 0
+        indx = np.argsort(np.ravel(self.xk))
+        self.xk = np.take(np.ravel(self.xk), indx, 0)
+        self.pk = np.take(np.ravel(self.pk), indx, 0)
+        self.a = self.xk[0]
+        self.b = self.xk[-1]
+        self.qvals = np.cumsum(self.pk, axis=0)
+
+        self.shapes = ' '   # bypass inspection
+        self._construct_argparser(meths_to_inspect=[self._pmf],
+                                  locscale_in='loc=0',
+                                  # scale=1 for discrete RVs
+                                  locscale_out='loc, 1')
+
+        self.vecentropy = self._entropy
+
+        # generate docstring for subclass instances
```

Comment:
From here till the end of ``__init__`` this is all duplicate code. And so is the 10 LoC starting from ``if badvalue is None:``. Can you keep that common? Then this ctor will only have the relevant changes from ``rv_discrete``.